eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -21,26 +21,14 @@ parameters:
     default: false
 
 steps:
-  - ${{ if eq(parameters.isWindows, 'true') }}:
-    - script: ${{ parameters.buildScript }}
-              -restore -build
-              -configuration ${{ parameters.buildConfig }}
-              /p:ArchiveTests=true
-              /bl:${{ parameters.repoLogPath }}/build.binlog
-              $(_OfficialBuildIdArgs)
-      displayName: Build
-  - ${{ if ne(parameters.isWindows, 'true') }}:
-    # temporary: Use -pack for linux as it is required for workload testing
-    # Issue: https://github.com/dotnet/aspire/issues/2687
-    - script: ${{ parameters.buildScript }}
-              -restore -build
-              -pack
-              -configuration ${{ parameters.buildConfig }}
-              /p:ArchiveTests=true
-              /bl:${{ parameters.repoLogPath }}/build.binlog
-              $(extraBuildArgs)
-              $(_OfficialBuildIdArgs)
-      displayName: Build
+  - script: ${{ parameters.buildScript }}
+            -restore -build
+            -pack
```

Comment:
Since we are now calling pack here always, we should probably start calling sign here too and remove the extra step we have later in this file called "Pack, Sign, and Publish". The original thinking of the split was that packacging and signing took longer so those should happen only after tests have validated things are good, but today they mostly just redo a bunch of stuff that we are already doing here, so might as well just merge the two.

Reply:
I want to tackle this one in a separate PR. Because for the tests I'm using a more constrained project `tests/workloads.proj` which results in faster installs. In a follow up I will check how to merge these two steps. Or probably be able to re-use parts of it.

Reply:
Sounds good to me, let's just make sure that we have an issue or something to track the de-duplication as that would save a bunch of build time.

Reply:
https://github.com/dotnet/aspire/issues/4222